### PR TITLE
ISSUE-461: Support multiple tags in Java plugin

### PR DESCRIPTION
**Pull request by codenet**

### CODE REVIEW

**Comment by codenet** [comment]
Thank you, Benjamin, for the reply!

The plan looks reasonable but it seems there's a lot more work for this PR to get merged. I'm not too familiar with the to-do list items you mentioned and am currently limited by the time I can invest in it. 

The build passes locally for current PR. Travis is complaining about DockerJavaApplicationPluginFunctionalTest."Can create image for Java application and push to private registry" which I wasn't able to reproduce locally.

Thanks again!

**Comment by bmuschko** [comment]
> The build passes locally for current PR. Travis is complaining about DockerJavaApplicationPluginFunctionalTest."Can create image for Java application and push to private registry" which I wasn't able to reproduce locally.

You'll only be able to run the test locally if you actually [set up a private repository](https://github.com/bmuschko/gradle-docker-plugin/blob/master/src/functTest/groovy/com/bmuschko/gradle/docker/DockerJavaApplicationPluginFunctionalTest.groovy#L255). If it doesn't exist then the test will be skipped, however, on CI we [create the private registry automatically](https://github.com/bmuschko/gradle-docker-plugin/blob/master/.travis.yml#L14). There's likely a problem in your code. Try to create that registry locally and run the tests. I will add the Gradle build scan functionality so we have more details about the failure.

File: src/main/groovy/com/bmuschko/gradle/docker/DockerJavaApplication.groovy [hunk 1]
```diff
@@ -18,8 +18,10 @@ package com.bmuschko.gradle.docker
 import com.bmuschko.gradle.docker.tasks.image.Dockerfile
 import groovy.transform.CompileStatic
 import groovy.transform.TypeCheckingMode
+import javafx.beans.property.ListProperty
```
Comment (bmuschko) [inline src/main/groovy/com/bmuschko/gradle/docker/DockerJavaApplication.groovy]: That's the wrong import.

**Comment by bmuschko** [comment]
You might actually be better off with writing your own plugin for your specific use cases. The plugin we have in place right now is fairly opinionated and likely won't make everyone happy, however, all capabilities are in place that make it easy to implement your own logic.

**Comment by bmuschko** [comment]
Just wondering...do you even care about the "push image" use case? If you don't then you can just add the image you want to the task by calling `dockerBuildImage.tags.add(...)`. That should do exactly what you want.

**Comment by bmuschko** [comment]
Just letting you know that I implemented the points 1, 2 and 4 above.

File: src/main/groovy/com/bmuschko/gradle/docker/DockerJavaApplicationPlugin.groovy [hunk 2]
```diff
@@ -189,23 +189,26 @@ class DockerJavaApplicationPlugin implements Plugin<Project> {
                     }))
                 }
                 // Can't be within `with` above because the compiler falls over.
-                dockerBuildImage.tag.set(determineImageTag(project, dockerJavaApplication))
+                dockerBuildImage.tags.set(determineImageTags(project, dockerJavaApplication))
```
[the remark this answers is not in the document]
Comment (bmuschko) [reply src/main/groovy/com/bmuschko/gradle/docker/DockerJavaApplicationPlugin.groovy]: Yeah, their change looks to be a bit inconsistent. They should have thought about the fact that users will also want to push multiple images as well.

**Comment by bmuschko** [comment]
I am going to close this PR for now. If we wanted to implement this functionality with what Docker Java has to offer then we'd have to create multiple push tasks if you provide more than one tag. If you just want to _build_ an image with multiple tags then you can do so by a simple one-liner as explained above. I'd suggest you follow up with the Docker Java project so they can add the capability to push multiple tags at once.

**Pull request closed without merging**